build/includes/allocation.mk

```diff
@@ -0,0 +1,79 @@
+# Copyright 2019 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+#         ____  ____   ____   _____           _ _	
+#    __ _|  _ \|  _ \ / ___| |_   _|__   ___ | (_)_ __   __ _	
+#   / _` | |_) | |_) | |       | |/ _ \ / _ \| | | '_ \ / _` |	
+#  | (_| |  _ <|  __/| |___    | | (_) | (_) | | | | | | (_| |	
+#   \__, |_| \_\_|    \____|   |_|\___/ \___/|_|_|_| |_|\__, |	
+#   |___/                                               |___/ 
+
+build_allocation_base_version = $(call sha,$(build_path)/build-sdk-images/tool/base/Dockerfile)
+build_allocation_base_tag = agones-build-allocation-base:$(build_allocation_base_version)
+
+# Calculate sha hash of sha hashes of all files in a specified ALLOCATION_FOLDER
+build_allocation_version = $(call sha_dir,$(build_path)/build-allocation-images/$(ALLOCATION_FOLDER)/*)
+build_allocation_base_remote_tag = $(REGISTRY)/$(build_allocation_base_tag)
+build_allocation_prefix = agones-build-allocation-
+grpc_release_tag = v1.16.1
+allocation_build_folder = build-allocation-images/
+ALLOCATION_FOLDER ?= go
+COMMAND ?= gen
+ALLOCATION_IMAGE_TAG=$(build_allocation_prefix)$(ALLOCATION_FOLDER):$(build_allocation_version)
+
+.PHONY: gen-all-allocation-grpc gen-allocation-grpc run-all-allocation-command run-allocation-command 
+
+# Generates grpc server and client for all supported languages.
+gen-all-allocation-grpc: COMMAND := gen
```

Comment:
Do we think we'll have multiple grpc generation pieces happening here?

I guess maybe?

Just wondering if we need all the machinery for multiple languages? What is your thoughts on this going forward?

Reply:
I can remove them for now and reintroduce when needed.

Reply:
I'm not against it - was just curious if you thought we would need it going forward or not.